docs/adr/0003-consensus-runtime-token-transfer.md

```diff
@@ -0,0 +1,460 @@
+}
+```
+
+**Fields:**
+
+- `beneficiary` specifies the beneficiary account address.
+- `amount_change` specifies the absolute value of the amount of base units to
+  change the allowance for.
+- `negative` specifies whether the `amount_change` should be subtracted instead
+  of added.
+
+The transaction signer implicitly specifies the general account. Upon executing
+the approval the following actions are performed:
+
+- If the `disable_allowances` staking consensus parameters is set to `true`,
+  the method fails with `ErrForbidden`.
+
+- The account indicated by the signer is loaded.
+
+- If the approval would create a new allowance and the maximum number of
+  allowances for an account has been reached, the method fails with
+  `ErrInvalidArgument`.
+
+- The set of approvals is updated so that the allowance is updated as specified
+  by `amount_change`/`negative`. In case the change would cause the allowance to
+  be equal to zero or negative, the allowance is removed.
+
+- The account is saved.
+
+- A corresponding `ApprovalEvent` is emitted with the following structure:
+
+  ```golang
+  type ApprovalEvent struct {
+      Owner       Address           `json:"owner"`
+      Beneficiary Address           `json:"beneficiary"`
+      Allowance   quantity.Quantity `json:"allowance"`
+  }
+  ```
+
+  Where `allowance` contains the new total allowance. The event is emitted even
+  if the new allowance is zero.
+
+#### Withdraw
+
+Withdraw enables a beneficiary to withdraw from the given account.
+
+**Method name:**
+
+```
+staking.Withdraw
+```
+
+**Body:**
+
+```golang
+type Withdraw struct {
+    From   Address           `json:"from"`
+    Amount quantity.Quantity `json:"amount"`
+}
+```
+
+**Fields:**
+
+- `from` specifies the account address to withdraw from.
+- `amount` specifies the amount of base units to withdraw.
+
+The transaction signer implicitly specifies the destination general account.
+Upon executing the withdrawal the following actions are performed:
+
+- If either the `disable_transfers` or `disable_allowances` staking consensus
+  parameters are set to `true`, the method fails with `ErrForbidden`.
+
+- Address specified by `from` is compared with the transaction signer address.
+  If the addresses are the same, the method fails with `ErrInvalidArgument`.
+
+- The source account indicated by `from` is loaded.
+
+- The destination account indicated by the transaction signer is loaded.
+
+- `amount` is deducted from the corresponding approval in the source account.
+  If this would cause the allowance to go negative, the method fails with
+  `ErrForbidden`.
+
+- `amount` is deducted from the source general account balance. If this would
+  cause the balance to go negative, the method fails with
+  `ErrInsufficientBalance`.
+
+- `amount` is added to the destination general account balance.
+
+- Both source and destination accounts are saved.
+
+- The corresponding `TransferEvent` is emitted.
```

Comment:
the allowance would change, without any event about it

Reply:
Good point, updated `ApprovalEvent` to `AllowanceChangeEvent` and made sure it is also emitted here.